doc/builders/packages/urxvt.xml

```diff
@@ -0,0 +1,90 @@
+<section xmlns="http://docbook.org/ns/docbook"
+         xmlns:xlink="http://www.w3.org/1999/xlink"
+         xml:id="sec-urxvt">
+ <title>Urxvt</title>
+
+ <para>
+  Urxvt, also known as rxvt-unicode, is a highly customizable terminal emulator.
+ </para>
+
+ <section xml:id="sec-urxvt-conf">
+
+  <title>Configuring urxvt</title>
+
+  <para>
+   In <literal>nixpkgs</literal>, urxvt is provided by the package
+   <literal>rxvt-unicode</literal>. It can be configured to include your choice
+   of plugins, reducing its closure size from the default configuration which
+   includes all available plugins. To make use of this functionality, use an
+   overlay or directly install an expression that overrides its configuration
+   such as
+   <programlisting>rxvt-unicode.override { configure = { availablePlugins, ... }: {
```

Comment:
I'm not sure I'm totally fond of this type of override. Why not go with the usual `with<Plugin>` scheme? I.e `withVtwheel` or `withResizefont`?

Reply:
I pretty much stole this design from the `weechat` package. You can see it in the nixpkgs [manual](https://nixos.org/nixpkgs/manual/#sec-weechat) It's a little verbose but it's completely configurable.

`withVtwheel` would require some kind of code generation or hard-coding a few combinations of plugins.

Reply:
On the same note of my confusion below, I tend to think it will be better...